addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java

```diff
@@ -0,0 +1,386 @@
+                    // CYC_FIREPLACE_TIMER (4613) = 0, checksum)
+                    HashMap<Integer, Integer> request = new HashMap<Integer, Integer>();
+                    request.put(4609, 1);
+                    request.put(4612, 0);
+                    request.put(4613, 0);
+                    return generateCustomRequest(249, request);
+                } else if (Integer.parseInt(updateState) == ValloxMVBindingConstants.STATE_BOOST) {
+                    // 30 Min boost (Length 6, Command to set data 249, CYC_BOOST_TIMER (4612) = 30, CYC_FIREPLACE_TIMER
+                    // (4613) = 0, checksum)
+                    HashMap<Integer, Integer> request = new HashMap<Integer, Integer>();
+                    request.put(4612, 30);
+                    request.put(4613, 0);
+                    return generateCustomRequest(249, request);
+                }
+            } else if (channelUID.getId().equals(ValloxMVBindingConstants.CHANNEL_ONOFF)) {
+                HashMap<Integer, Integer> request = new HashMap<Integer, Integer>();
+                request.put(4610, Integer.parseInt(updateState));
+                return generateCustomRequest(249, request);
+            } else if (channelUID.getId().equals(ValloxMVBindingConstants.CHANNEL_EXTR_FAN_BALANCE_BASE)) {
+                HashMap<Integer, Integer> request = new HashMap<Integer, Integer>();
+                request.put(20485, Integer.parseInt(updateState));
+                return generateCustomRequest(249, request);
+            } else if (channelUID.getId().equals(ValloxMVBindingConstants.CHANNEL_SUPP_FAN_BALANCE_BASE)) {
+                HashMap<Integer, Integer> request = new HashMap<Integer, Integer>();
+                request.put(20486, Integer.parseInt(updateState));
+                return generateCustomRequest(249, request);
+            } else if (channelUID.getId().equals(ValloxMVBindingConstants.CHANNEL_HOME_SPEED_SETTING)) {
+                HashMap<Integer, Integer> request = new HashMap<Integer, Integer>();
+                request.put(20507, Integer.parseInt(updateState));
+                return generateCustomRequest(249, request);
+            } else if (channelUID.getId().equals(ValloxMVBindingConstants.CHANNEL_HOME_AIR_TEMP_TARGET)) {
+                HashMap<Integer, Integer> request = new HashMap<Integer, Integer>();
+                request.put(20508, Integer.parseInt(updateState));
+                return generateCustomRequest(249, request);
+            }
+            return null;
+        }
+
+        @OnWebSocketMessage
+        public void onMessage(String message) {
+            logger.debug("Message from Server: {}", message);
+        }
+
+        @OnWebSocketError
+        public void onError(Throwable cause) {
+            voHandler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            logger.debug("Error: {}", cause.getMessage().toString());
```

Comment:
Please provide context and note that you don't have to call `toString` it will be called automatically, but only when the logging-level actually applies.

Reply:
Changed